src/utils/namestone.ts

```diff
@@ -0,0 +1,165 @@
+import { logError } from "./errors";
+
+// Namestone API types
+export type NamestoneSetNameRequest = {
+  name: string;
+  domain: string;
+  address: string;
+  text_records?: Record<string, string>;
+  coin_types?: Record<string, string>;
+};
+
+export type NamestoneDeleteNameRequest = {
+  name: string;
+  domain: string;
+};
+
+export type NamestoneResponse = {
+  success: boolean;
+  message?: string;
+  error?: string;
+};
+
+class NamestoneService {
+  private readonly apiKey: string;
+  private readonly domain: string;
+  private readonly baseUrl: string;
+
+  constructor() {
+    this.apiKey = process.env.NAMESTONE_API_KEY || "";
+    this.domain = process.env.NAMESTONE_DOMAIN || "";
+    this.baseUrl =
+      process.env.NAMESTONE_BASE_URL || "https://namestone.com/api/public_v1";
+
+    if (!this.apiKey) {
+      console.warn(
+        "NAMESTONE_API_KEY not configured - Namestone integration disabled",
+      );
+    }
+    if (!this.domain) {
+      console.warn(
+        "NAMESTONE_DOMAIN not configured - Namestone integration disabled",
+      );
+    }
+  }
+
+  private isEnabled(): boolean {
+    return !!(this.apiKey && this.domain);
+  }
+
+  /**
+   * Sets a name (subdomain) for a given address and domain
+   */
+  async setName(args: {
+    username: string;
+    address: string;
+    textRecords?: Record<string, string>;
```

Comment:
Maybe we can type this better. Like the textRecords, can they key be anything? or only some specific ones?